src/plugins/intel_gpu/src/graph/program_node.cpp

```diff
@@ -268,6 +268,15 @@ bool program_node::recalc_output_layout(bool invalidate_users_if_changed) {
     return set_output_layout(new_layout, invalidate_users_if_changed);
 }
 
+bool program_node::is_dynamic() const {
+    for (auto& input : get_dependencies()) {
+        if (input->get_output_layout().is_dynamic())
```

Comment:
Are you trying to avoid recursive call here by calling layout's `is_dynamic()` instead of node, right? I don't fully understand how layouts will be marked as dynamic... But if they will be marked at initialization stage than current approach is ok
But is it not possible that in the middle of graph will be created layout which will be marked as static for some reason, and than we need something like recursive call for correct type propagation?

Reply:
Layouts are not marked by any step. PartialShape may always say if it's dynamic or static based on internal dimensions